packages/server/src/queries/complex/pools/index.ts

```diff
@@ -91,72 +117,20 @@ export async function getPool({
 export async function getPools(
   params: Partial<PoolFilter> & { assetLists: AssetList[]; chainList: Chain[] },
   poolProvider: PoolProvider = getPoolsFromSidecar
-): Promise<Pool[]> {
-  let pools = await poolProvider(params);
-
-  pools = pools.filter((pool) => !FILTERABLE_IDS.includes(pool.id)); // Filter out ids in FILTERABLE_IDS
-
-  if (params?.types) {
-    pools = pools.filter(({ type }) =>
-      params?.types ? params.types.includes(type) : true
-    );
-  }
-
-  // Note: we do not want to filter the pools if we are in testnet because we do not have accurate pricing
-  // information.
-  if (params?.minLiquidityUsd && !IS_TESTNET) {
-    pools = pools.filter(({ totalFiatValueLocked }) =>
-      params?.minLiquidityUsd
-        ? totalFiatValueLocked.toDec().gte(new Dec(params.minLiquidityUsd))
-        : true
-    );
-  }
-
-  // add denoms so user can search them
-  let denomPools = pools.map((pool) => ({
-    ...pool,
-    coinDenoms: pool.reserveCoins.flatMap((coin) => [
-      coin.denom,
-      coin.currency.coinMinimalDenom,
-    ]),
-    poolNameByDenom: pool.reserveCoins.map(({ denom }) => denom).join("/"),
-    coinNames: pool.reserveCoins.map((coin) => [
-      // @ts-ignore
-      coin.currency.coinName,
-    ]),
-  }));
-
-  const denoms = params.denoms;
-  if (denoms) {
-    denomPools = denomPools.filter((denomPool) =>
-      denomPool.coinDenoms.some((denom) => denoms.includes(denom))
-    );
-  }
-
-  if (params?.search) {
-    // search for an exact match of coinMinimalDenom or pool ID
-    const coinDenomsOrIdMatches = search(
-      denomPools,
-      ["coinDenoms", "id"],
-      params.search,
-      0.0 // Exact match
-    );
-
-    // if not exact match for coinMinimalDenom or pool ID, search by poolNameByDenom (ex: OSMO/USDC) or coinName (ex: Bitcoin)
-    if (coinDenomsOrIdMatches.length > 0) {
-      denomPools = coinDenomsOrIdMatches;
-    } else {
-      const poolNameByDenomMatches = search(
-        denomPools,
-        ["poolNameByDenom", "coinNames"],
-        params.search
-      );
-
-      denomPools = poolNameByDenomMatches;
-    }
-  }
-
-  return denomPools;
+): Promise<{
+  items: Pool[];
+  total: number;
+  nextCursor: number | undefined;
+}> {
+  params.notPoolIds = [...(params.notPoolIds ?? []), ...FILTERABLE_IDS];
```

Comment:
Maybe we can just do here:
```ts
params.notPoolIds = !params.notPoolIds ? FILTERABLE_IDS : params.notPoolIds
```

This ensure we set `FILTERABLE_IDS` as default value, but then you can totally override it, does it makes sense?

Reply:
Yea, it makes sense! Adjusted as per you suggestion, thank you!